api/src/test/java/com/epam/pipeline/controller/configuration/ConfigurationControllerTest.java

```diff
@@ -0,0 +1,138 @@
+
+import com.epam.pipeline.config.JsonMapper;
+import com.epam.pipeline.controller.vo.configuration.RunConfigurationVO;
+import com.epam.pipeline.entity.configuration.RunConfiguration;
+import com.epam.pipeline.manager.configuration.RunConfigurationApiService;
+import com.epam.pipeline.test.creator.configuration.ConfigurationCreatorUtils;
+import com.epam.pipeline.test.web.AbstractControllerTest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+@WebMvcTest(controllers = ConfigurationController.class)
+public class ConfigurationControllerTest extends AbstractControllerTest {
+
+    private static final long ID = 1L;
+    private static final String CONFIGURATION_URL = SERVLET_PATH + "/configuration";
+    private static final String CONFIGURATION_BY_ID_URL = CONFIGURATION_URL + "/%d";
+    private static final String ALL_CONFIGURATIONS_URL = CONFIGURATION_URL + "/loadAll";
+    private JsonMapper mapper;
+    private final RunConfiguration runConfiguration = ConfigurationCreatorUtils.getRunConfiguration();
+    private final RunConfigurationVO runConfigurationVO = ConfigurationCreatorUtils.getRunConfigurationVO();
+
+    @Autowired
+    private RunConfigurationApiService mockRunConfigurationApiService;
+
+    @Before
+    public void setUp() {
+        mapper = getObjectMapper();
+        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    }
+
+    @Test
+    public void shouldFailSaveConfigurationForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(post(CONFIGURATION_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldSaveConfiguration() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(runConfigurationVO);
+        Mockito.doReturn(runConfiguration).when(mockRunConfigurationApiService).save(Mockito.refEq(runConfigurationVO));
+
+        final MvcResult mvcResult = performRequest(post(CONFIGURATION_URL).content(content));
+
+        Mockito.verify(mockRunConfigurationApiService).save(Mockito.refEq(runConfigurationVO));
+        assertResponse(mvcResult, mapper, runConfiguration, ConfigurationCreatorUtils.RUN_CONFIGURATION_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUpdateConfiguration() throws Exception {
+        runConfigurationVO.setId(ID);
```

Comment:
You have to create an additional instance of run configuration. Otherwise the test class will fail once tests execution order is different which is possible since junit 4 doesn't have any guarantees on tests execution order.